### ISSUE TITLE
MaskedDataset incorrectly masks padding tokens when is_rna=True

### ISSUE THREAD

**Issue by agam263** [gc-os-ai/pyaptamer#592]
# [BUG]: RNA adjacent masking corrupts padding tokens in MaskedDataset

### Description
When `MaskedDataset` is initialized with `is_rna=True`, it implements a strategy to mask nucleotides adjacent to randomly sampled positions (to account for base pairing in RNA). However, the current implementation in `_mask_rna` does not verify if the adjacent positions contain valid nucleotides or are simply padding tokens (`0`).

In sequences that are padded to `max_len`, this causes the mask token (`mask_idx`) to overwrite padding tokens at the end of the sequence.

### Technical Root Cause
In `pyaptamer/datasets/dataclasses/_masked.py`, the `_mask_rna` method blindly appends `pos + 1` and `pos - 1` to the list of masked positions:

```python
for pos in mask_positions:
    if pos < self.max_len - 1:
        adjacent_positions.append(pos + 1) # <--- BUG: This position may be 0 (padding)
```

Additionally, in `__getitem__`, these "extra" masked positions are not removed from the `no_mask_positions` list, which causes them to be zeroed out in the target labels (`y_masked`), making it impossible for the model to learn from these specifically masked tokens.

### Impact
1.  **Corrupted Padding**: The model is forced to perform Masked Language Modeling (MLM) on empty padding slots, which introduces noise and slows down convergence.
2.  **Broken Training Signal**: Since the adjacent masked tokens aren't correctly handled in the target labels, the model is penalized for predicting anything other than `0` (padding) at those positions, contradicting the purpose of the RNA-specific masking strategy.

### Reproduction Steps
```python
from pyaptamer.datasets.dataclasses import MaskedDataset
import torch

# Sequence of length 3, padded to 5
sequences = [[1, 2, 3, 0, 0]]
targets = [[1, 2, 3, 0, 0]]
dataset = MaskedDataset(sequences, targets, max_len=5, mask_idx=5, masked_rate=0.5, is_rna=True)

# If index 2 is picked for masking, index 3 (padding) will also be masked
x_masked, y_masked, _, _ = dataset[0]
print(f"Masked Sequence: {x_masked.tolist()}") 
# Actual Output: [1, 2, 5, 5, 0]  <-- Index 3 is now a mask token instead of padding!
```

### Proposed Fix
1.  Add a check in `_mask_rna` to ensure `x_masked[pos ± 1] > 0` before masking.
2.  Update the `__getitem__` logic to ensure that tokens masked by the RNA strategy are included in the target labels and excluded from the "zeroing out" phase.

***

**Comment by pranavchoudhary-tech** [comment]
Hi! I'd like to work on this issue as part of my European Summer of Code application.

I looked into the `MaskedDataset` logic and saw how `_mask_rna` was blindly appending padding tokens (`0`). I have updated the RNA masking logic to strictly check for `> 0` adjacent tokens and ensured they aren't incorrectly zeroed out during the `__getitem__` sampling phase. 

I have the code ready and will bundle it into my upcoming PR.

**Comment by agam263** [comment]
@pranavchoudhary-tech Hi, thanks for your effort on this!
I’ve already resolved the issue, and this PR is currently under review by the mentor.
I really appreciate your contribution.
